From 92e7b5189238453aed25a66031252c1d03464cf4 Mon Sep 17 00:00:00 2001
From: Mohan Thadikamalla <mohant@nvidia.com>
Date: Fri, 9 Jun 2017 14:03:16 +0530
Subject: [PATCH 21/28] frameworks/base: Wifi application property support

feature: NvPropSupport
Bug 200316146

In order to optimize the wifi scan algorithms, wpa_supplicant
needs to know the state of the applications. So add wifi
application property WiFi Manager.

Bug 1400711

Change-Id: I8d434e1b9fe785b4a619d1faafe0482d89a334fe
Reviewed-on: http://git-master/r/1212008
(cherry picked from commit cb26f117631173976e22d66fe7e94567d077fe73)
Reviewed-on: http://git-psac/r/10412
Reviewed-by: Mohan Thadikamalla <mohant@nvidia.com>
Tested-by: Mohan Thadikamalla <mohant@nvidia.com>
Reviewed-by: Srinivas Ramachandran <srinivasra@nvidia.com>
Reviewed-by: David Pearce <dpearce@nvidia.com>
Tested-by: David Pearce <dpearce@nvidia.com>
Reviewed-on: http://git-psac.nvidia.com/r/13451
GVS: Gerrit_Virtual_Submit
Reviewed-by: Bibhay Ranjan <bibhayr@nvidia.com>
Tested-by: Bibhay Ranjan <bibhayr@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/1552641
Reviewed-by: Niranjan Artal <nartal@nvidia.com>
Tested-by: Niranjan Artal <nartal@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/1585021
---
 wifi/java/android/net/wifi/IWifiManager.aidl |  6 +++
 wifi/java/android/net/wifi/WifiManager.java  | 46 ++++++++++++++++++++
 2 files changed, 52 insertions(+)
 mode change 100644 => 100755 wifi/java/android/net/wifi/IWifiManager.aidl

diff --git a/wifi/java/android/net/wifi/IWifiManager.aidl b/wifi/java/android/net/wifi/IWifiManager.aidl
old mode 100644
new mode 100755
index a3a1054f869..3250916e00a
--- a/wifi/java/android/net/wifi/IWifiManager.aidl
+++ b/wifi/java/android/net/wifi/IWifiManager.aidl
@@ -184,5 +184,11 @@ interface IWifiManager
     void restoreBackupData(in byte[] data);
 
     void restoreSupplicantBackupData(in byte[] supplicantData, in byte[] ipConfigData);
+
+    String getAppProperty(String key);
+
+    int setAppProperty(String key, String value);
+
+    int setAppPropertyOnDeath(IBinder binder, String key, String value);
 }
 
diff --git a/wifi/java/android/net/wifi/WifiManager.java b/wifi/java/android/net/wifi/WifiManager.java
index 9c8ea88c8c8..13555fb3444 100644
--- a/wifi/java/android/net/wifi/WifiManager.java
+++ b/wifi/java/android/net/wifi/WifiManager.java
@@ -3603,4 +3603,50 @@ public class WifiManager {
             throw e.rethrowFromSystemServer();
         }
     }
+
+    /**
+     * Class to access wifi application properties
+     * @hide
+     */
+    public class WifiAppPropertyManager {
+        private final IBinder mBinder;
+
+        private WifiAppPropertyManager() {
+            mBinder = new Binder();
+        }
+
+        public String getProperty(String key) {
+            try {
+                return mService.getAppProperty(key);
+            } catch (RemoteException e) {
+                return null;
+            }
+        }
+
+        public int setProperty(String key, String value) {
+            try {
+                return mService.setAppProperty(key, value);
+            } catch (RemoteException e) {
+                return -1;
+            }
+        }
+
+        public int setPropertyOnDeath(String key, String value) {
+            synchronized (mBinder) {
+                try {
+                    return mService.setAppPropertyOnDeath(mBinder, key, value);
+                } catch (RemoteException e) {
+                    return -1;
+                }
+            }
+        }
+    }
+
+    /**
+     * API to get wifi application property manager
+     * @hide
+     */
+    public WifiAppPropertyManager getWifiAppPropertyManager() {
+        return new WifiAppPropertyManager();
+    }
 }
-- 
2.17.1

